### DISCUSSION TITLE
Splitting tokenization part to reduce time

### DISCUSSION THREAD

**Discussion by yusufcakmakk** [ymcui/Chinese-LLaMA-Alpaca-2#432] [Ideas]
Hi all,

I have a suggestion for the tokenization part in [here](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/blob/main/scripts/training/run_clm_pt_with_peft.py#L523). When we start the process, and we have different block sizes, it calculates all tokens again and then grouping them. Instead, the tokenization step can be used once and then grouped based on block size. 

What do you think about this improvement?

**Comment by iMountTai** [comment]
This can be achieved by modifying [here](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/blob/0189e8be7706d2aceb90d238149d5fda6b6aea8d/scripts/training/run_clm_pt_with_peft.py#L530C33-L530C33).
